### PR TITLE
Image handler

### PR DESCRIPTION
**Pull request by domanttas**
Added wrapper for REST Service class;
Managed to get variable of type Image from URI in Azure Cloud;
Moved checking for permission to another class in static methods (not sure if it's considered good practise);
Moved taking of photo logic to another handler class, same case with static ??;

Notes: will update these handlers with options to get list of Images from URIs which could be feeded to image recognition API.

### CODE REVIEW

File: WhoIsThat/WhoIsThat/Handlers/TakingPhotoHandler.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                    throw new ArgumentException("Photo was not successfully taken", "MediaFile");
+                }
+
+                else
+                {
+                    return file;
+                }
+            }
+
+            catch (Exception exception)
```
Comment (tomasmuzas) [inline WhoIsThat/WhoIsThat/Handlers/TakingPhotoHandler.cs]: As I have been taught, if you are catching `System.Exception` think two or three times before doing that :D especially, when you are just rethrowing it

File: WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs [hunk 2]
```diff
@@ -24,44 +26,26 @@ public HomePage ()
...
-                //Not sure if we should log or display caught exception, gotta figure it out
-                await DisplayAlert("Something went wrong", "Please try again", "OK");
-            }
+                var stream = takenPhoto.GetStream();
+                takenPhoto.Dispose();
+                return stream;
+            });
         }
     }
 }
```
Comment (tomasmuzas) [inline WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs]: Thats a good example of refactoring!

File: WhoIsThat/WhoIsThat/Handlers/ImageHandler.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        /// <summary>
+        /// Creates Image from provided URI (which is present in ImageObject)
+        /// </summary>
+        /// <param name="uri">URI of image in Azure cloud</param>
+        /// <returns>Variable of type Image which contains Source of image in Azure cloud</returns>
+        public Image GetImageFromUri(string uri)
+        {
+            var fetchedImage = new Image
+            {
+                Source = ImageSource.FromUri(new Uri(uri))
```
[the remark this answers is not in the document]
Comment (domanttas) [reply WhoIsThat/WhoIsThat/Handlers/ImageHandler.cs]: If it's an rhetorical question, it's a bit of a mystery to me as well, so I have no clue how to change it. I have tested it and I successfully displayed image from Azure Cloud using this method. Then again, I'm not sure about anything.

File: WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs (hunk 2 above)
Comment (domanttas) [reply WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs]: As always, thank you for your feedback! Really appreciate your insights.

File: WhoIsThat/WhoIsThat/Handlers/ImageHandler.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomasmuzas) [reply WhoIsThat/WhoIsThat/Handlers/ImageHandler.cs]: Well, it wasnt supposed to be rhetorical, but thats an interesting choice of implementation by MS...

File: WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs (hunk 2 above)
Comment (tomasmuzas) [reply WhoIsThat/WhoIsThat/Views/HomePage.xaml.cs]: No problem! Take a look at `Autofac` lib, maybe you will be able to put it in use

**Pull request merged**